### DISCUSSION TITLE
npm init next-app fails on Android with Termux

### DISCUSSION THREAD

**Discussion by warp9pnt9** [vercel/next.js#15252] [Help]
OS: Android 5.0 with Termux v0.83
Node: v12.13.0
npm: 6.14.6

Same error given for ```npx create-react-app``` and ```npx create-next-app```.  Help me understand the cause of the problem and what to do to fix this.

```
$ npm init next-app cp-c19t
npx: installed 1 in 20.883s
? Pick a template › - Use arrow-keys. Return to ✔ Pick a template › Default starter app
Creating a new Next.js app in /data/data/com.termux/files/home/dev/cp-c19t.

Installing react, react-dom, and next using npm...

events.js:187
      throw er; // Unhandled 'error' event
      ^

Error: spawn npm ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:264:19)
    at onErrorNT (internal/child_process.js:456:16)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (internal/child_process.js:270:12)
    at onErrorNT (internal/child_process.js:456:16)
    at processTicksAndRejections (internal/process/task_queues.js:80:21) {
  errno: 'ENOENT',
  code: 'ENOENT',
  syscall: 'spawn npm',
  path: 'npm',
  spawnargs: [
    'install',
    '--save',
    '--save-exact',
    '--loglevel',
    'error',
    'react',
    'react-dom',
    'next'
  ]
}

```

**Comment by CodeWithLinc** [comment]
I really don't know the cause of this error but there is a way to fix it.

My reasoning for the error: the npm installed via "pkg install nodejs" is very compressed or it need root access (the last part of the log.txt message).

NOTE: download the latest Termux then update and upgrade it via
-pkg update
-pkg upgrade (say Y "YES" to all the prompts)

The Solution
Download Andronix
https://play.google.com/store/apps/details?id=studio.com.techriz.andronix

install one of the distros
(I installed Ubuntu 20.04 - any version doesn't matter - 18.04 or 20.04)

Run these commands when you are at root@localhost
-apt-get update
-apt-get upgrade
-apt-get install nodejs
apt-get install npm

👆👆👆That's all

Now build your Next.js Applications with "npx create-next-app <APPLICATION NAME>"
![Screenshot_20210125-070734](https://user-images.githubusercontent.com/55532988/105666227-7778f380-5ee1-11eb-9345-bc32dd601d5a.png)

NOTE:  👆👆👆 The next app will be build in the "Ubuntu(version) fs" folder in the termux directory (I used the default DOWNLOAD application to view this)
![Screenshot_20210125-074944](https://user-images.githubusercontent.com/55532988/105666595-4816b680-5ee2-11eb-8eb4-30dd33582902.png)

My Recommendations NOT a must
-Enter the "Ubuntu(version) folder
-Create your application in the "sdcard" folder like below
![Screenshot_20210125-075004](https://user-images.githubusercontent.com/55532988/105666722-9035d900-5ee2-11eb-9a20-49514de14f54.png)

![Screenshot_20210125-075022](https://user-images.githubusercontent.com/55532988/105666752-9fb52200-5ee2-11eb-913e-22e4949b863a.png)

EDITOR
Acode Editor (Use any, the premium or free version)
Free Version
https://play.google.com/store/apps/details?id=com.foxdebug.acodefree

Premium Version
https://play.google.com/store/apps/details?id=com.foxdebug.acode

Add the Termux directory to the folder paths in the Acode editor
Run "npm run dev" or other scripts via termux
Edit the files using Acode
😃😃😃😃
All the best

**Reply by md-fahad-ali32** [reply]
Not working

**Comment by CodeWithLinc** [comment]
By the way, don't forget to run this command after you install, upgrade and update Termux

termux-setup-storage 
👆👆👆
(A prompt for File and Media access by Termux)

**Reply by md-fahad-ali32** [reply]
> By the way, don't forget to run this command after you install, upgrade and update Termux
> 
> termux-setup-storage 👆👆👆 (A prompt for File and Media access by Termux)

I'ts doesn't work

**Reply by LincCodes** [reply]
> > By the way, don't forget to run this command after you install, upgrade and update Termux
> > 
> > termux-setup-storage 👆👆👆 (A prompt for File and Media access by Termux)
> 
> I'ts doesn't work

Hey bro, you DON'T need to install any distro, you can just

Download termux
Update and upgrade it
Install node
install nextjs
Finally start a new next app

All this WITHOUT installing any linux distro.

**Reply by md-fahad-ali32** [reply]
Thanks for suggestion it install a dependency and that is eslint-config-next i can't install that it's shows an error ...'
i am using armv7l prcessor is it work?
